apps/src/calc/inputIterator.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Given a set of values (i.e. [1,2,3], and a number of parameters, generates
+ * all possible combinations of values.
+ */
+var InputIterator = function (values, numParams) {
+  this.numParams_ = numParams;
+  this.remaining_ = Math.pow(values.length, numParams);
+  this.values_ = values;
```

Comment:
Maybe s/`values`/`availableValues` to differentiate "bucket of available ones" from the current/next generated permutations